stock_force_assign_by_type/models/stock_move.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Tecnativa S.L. - Pilar Vargas
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _should_bypass_reservation(self, forced_location=False):
+        res = super()._should_bypass_reservation()
+        if self.picking_type_id.force_reservation:
+            return False
+        return res
```

Comment:
No need of calling super in the case of forcing.

```suggestion
        if self.picking_type_id.force_reservation:
            return False
        return super()._should_bypass_reservation()
```